src/app/domain/robots.ts

```diff
@@ -195,3 +195,102 @@ export const arduinoMegaRobotType = new RobotType(
         showLeaphySensors: true,
     },
 );
+
+export const arduinoMegaRobotTypeGeneric = new RobotType(
```

Comment:
Why duplicate this? The libs are only installed, not included, and they are cached anyways. Also, if you switch from blocks code to advanced and use compass or apds blocks, your code suddenly doesn't compile because the libs are missing and have to be manually installed.
Lastly, when switching from blocks to advanced, the wrong robot will be selected .

Reply:
Because otherwise other states break, for example the canChangeCodeEditor

Reply:
canChangeCodeEditor shouldn't depend on robotType, that makes no sense to me... I see we do this now because of the 'are you sure' when switching back to blocks, but that could be better solved by adding a 'supportsBlocks' bool to the robotType and checking that. We really need to get rid of all these hard-coded lists of robots/robotTypes and a million checks everywhere.